src/cmd/usb-util.js

```diff
@@ -134,6 +137,33 @@ async function executeWithUsbDevice({ args, func, dfuMode = false } = {}) {
 	return res;
 }
 
+/**
+ * Reboots device and waits for it to enter normal mode.
+ * Useful for enabling Device Protection on a device in after its current operation completes.
+ * @param {*} deviceId
+ * @returns
+ */
+async function waitForDeviceToReboot(deviceId) {
+	const REBOOT_TIME_MSEC = 60000;
+	const REBOOT_INTERVAL_MSEC = 1000;
+	const start = Date.now();
+	while (Date.now() - start < REBOOT_TIME_MSEC) {
+		try {
+			await _delay(REBOOT_INTERVAL_MSEC);
+			const device = await reopenDevice({ id: deviceId });
+			// Check device readiness
+			await device.getDeviceId();
+			return device;
+		} catch (error) {
+			// ignore error
+		}
+	}
+}
+
+async function _delay(ms){
```

Comment:
We already have a delay function 👉 https://github.com/particle-iot/particle-cli/blob/0fcc1d661ad7331d20eb9578a798a20cfcfe2f3b/src/lib/utilities.js#L139 Can you use it instead of creating a new one?